### PR TITLE
transport_sockets: removed well_known_names.h file.

### PR DESCRIPTION
**Pull request by cpakulski**
Commit Message:
Removed transport_sockets/well_known_names.h file.

Additional Description:
This is as per comment here: https://github.com/envoyproxy/envoy/pull/13112#discussion_r540589046

Risk Level: Low
Testing: Make sure that all unit and integration tests pass
Docs Changes: No
Release Notes: No
Platform Specific Features:
Partially fixes: #7238

### CODE REVIEW

**Comment by asraa** [comment]
For x-ref, this is the issue Matt was referring to https://github.com/envoyproxy/envoy/issues/7238

**Comment by cpakulski** [comment]
Thanks @asraa for pointing to the original issue. I will update the PR description with the issue number.

File: source/common/upstream/upstream_impl.cc [hunk 1]
```diff
@@ -867,12 +866,11 @@ Network::TransportSocketFactoryPtr createTransportSocketFactory(
   auto transport_socket = config.transport_socket();
   if (!config.has_transport_socket()) {
     if (config.has_hidden_envoy_deprecated_tls_context()) {
-      transport_socket.set_name(Extensions::TransportSockets::TransportSocketNames::get().Tls);
+      transport_socket.set_name("envoy.transport_sockets.tls");
```
Comment (asraa) [inline source/common/upstream/upstream_impl.cc:869]: I think the changes to adding the names inline in `name()` methods in source/extensions make sense to me. Here and in `source/server`, references to tls/raw_buffer names seem like they should refer to a common string, but it doesn't seem like there's a good place for that unless it exists in `common/network` somewhere.
@mattklein123 @htuch do you have ideas or should this just stay as is? 
If it should stay as is, PR LGTM.

**Pull request closed without merging**